imager_images/forms.py

```diff
@@ -1,47 +1,191 @@
+        fields = ['image',
+                  'title',
+                  'description',
+                  'published',
+                  ]
 
-class EditPhotoForm(ModelForm):
     def __init__(self, *args, **kwargs):
-        super(EditPhotoForm, self).__init__(*args, **kwargs)
-        self.fields['albums'].queryset = self.instance.user.albums.all()
+        self.request = kwargs.pop('user', None)
+        return super(PhotoAddViewForm, self).__init__(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        kwargs['commit'] = False
+        obj = super(PhotoAddViewForm, self).save(*args, **kwargs)
+        if self.request:
+            obj.user = self.request
+        obj.save()
+        return obj
 
+
+class PhotoUpdateViewForm(ModelForm):
     class Meta:
         model = Photo
+        widgets = {'image': ImageWidget(template='%(image)s<br />')}
+        fields = ('image',
+                  'albums',
+                  'title',
+                  'description',
+                  'published',
+                  )
+
+    def __init__(self, *args, **kwargs):
+        super(PhotoUpdateViewForm, self).__init__(*args, **kwargs)
+        self.fields['albums'].queryset = self.instance.user.albums.all()
+
+    def save(self, *args, **kwargs):
+        kwargs['commit'] = False
+
+        obj = super(PhotoUpdateViewForm, self).save(*args, **kwargs)
+        if (obj.published == u'pub' or obj.published == u'shd') and not obj.date_published:
+            obj.date_published = datetime.datetime.utcnow()
+        else:
+            obj.date_published = None
+
+        obj.save()
+        self.save_m2m()
+
+
+# class PhotoDeleteViewForm(ModelForm):
+#     class Meta:
+#         model = Photo
+#         widgets = {'image': ImageWidget(template='%(image)s<br />')}
+#         fields = ('image',
+#                   )
+
+#     def __init__(self, *args, **kwargs):
+#         super(PhotoDeleteViewForm, self).__init__(*args, **kwargs)
+
+
+# class AlbumUpdateViewForm(ModelForm):
+#     def __init__(self, *args, **kwargs):
+#         super(AlbumUpdateViewForm, self).__init__(*args, **kwargs)
+#         # import ipdb; ipdb.set_trace()
+#         self.fields['photos'].queryset = self.instance.user.photos.all()
+
+#     class Meta:
+#         model = Album
+
+#         fields = ('title',
+#                   'description',
+#                   'photos',
+#                   'published'
+#                   )
+
+
+# class PhotoAlbumForm(ModelForm):
+#     def __init__(self, *args, **kwargs):
+#         super(PhotoAlbumForm, self).__init__(*args, **kwargs)
+
+#         # # self.fields['photo'].queryset.filter(user=self.instance.album.user)
+#         import pdb; pdb.set_trace()
+#         try:
+#             self.instance.album
+#         except:
+#             pass
+#         else:'photo'].queryset = self.fields['photo'].queryset.filter(
+#                 user=self.instance.album.user)
+
+#     class Meta:
+#          model = Photo.albums.through
```

Comment:
Why all the commented out code?  Just delete it.  It's in git, you can get it back if you need.